graphql_compiler/tests/test_schema_fingerprint.py

```diff
@@ -64,7 +65,172 @@ def test_schema_fingerprint_basic(self):
         )
         self.assertNotEqual(schema_with_added_field_fingerprint, fingerprint)
 
-    def test_interface_order(self):
+    def test_different_field_type(self):
+        schema_text1 = """
+            type Object {
+                field: String
+            }
+        """
+        schema_text2 = """
+            type Object {
+                field: Int
+            }
+        """
+        _compare_schema_fingerprints(self, schema_text1, schema_text2, expect_equality=False)
+
+    def test_field_argument_order(self):
+        schema_text1 = """
+            type Object {
+                field(a: Int, b: String): String
+            }
+        """
+        schema_text2 = """
+            type Object {
+                field(b: String, a: Int): String
+            }
+        """
+        _compare_schema_fingerprints(self, schema_text1, schema_text2)
+
+    def test_different_field_arguments(self):
+        schema_text1 = """
+            type Object {
+                field(a: Int, b: String): String
+            }
+        """
+        schema_text2 = """
+            type Object {
+                field(a: Int): Int
+            }
+        """
+        _compare_schema_fingerprints(self, schema_text1, schema_text2, expect_equality=False)
+
+    def test_enum_order(self):
+        schema_text1 = """
+            enum Enum{
+                A
+                B
+            }
+        """
+        schema_text2 = """
+            enum Enum{
+                B
+                A
+            }
+        """
+        _compare_schema_fingerprints(self, schema_text1, schema_text2)
+
+    def test_different_enum(self):
+        schema_text1 = """
+            enum Enum{
+                A
+                B
+            }
+        """
+        schema_text2 = """
+            enum Enum{
+                A
+            }
+        """
+        _compare_schema_fingerprints(self, schema_text1, schema_text2, expect_equality=False)
+
+    def test_input_order(self):
+        schema_text1 = """
+            input Input {
+                a: Int
+                b: Int
+            }
+        """
+        schema_text2 = """
+            input Input {
+                b: Int
+                a: Int
+            }
+        """
+        _compare_schema_fingerprints(self, schema_text1, schema_text2)
+
+    def test_different_input(self):
+        schema_text1 = """
+            input Input {
+                a: Int
+                b: Int
+            }
+        """
+        schema_text2 = """
+            input Input {
+                a: Int
+            }
+        """
+        _compare_schema_fingerprints(self, schema_text1, schema_text2, expect_equality=False)
+
+    @pytest.mark.xfail(
+        strict=True,
+        reason="Type extension information is lost when a schema is print and parsed. "
+        "See https://github.com/graphql/graphql-js/issues/2386",
```

Comment:
nit: I think it might be a bit more readable if you surround the string literal here in parens: `reason=(...)`